### PR TITLE
travis-ci: refactor package management, i.e. use travis-ci "apt" plugin, cache packages

### PR DESCRIPTION
**Pull request by chipitsine**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -107,6 +115,7 @@ script:
   - cd ..
   - export PATH=$PWD/work/nginx/sbin:$PWD/nginx-devel-utils:$PATH
   - export NGX_BUILD_CC=$CC
+  - if [ "$CC" == "clang" ]; then export CFLAGS="-Qunused-arguments"; fi
```
Comment (agentzh) [inline .travis.yml]: Will you document why we need this? Looks strange.

File: .travis.yml [hunk 2]
```diff
@@ -49,10 +60,8 @@ services:
  - memcache
  - redis-server
 
-before_install:
-  - sudo apt-get install -y axel cpanminus libtest-base-perl libtext-diff-perl liburi-perl libwww-perl libtest-longstring-perl liblist-moreutils-perl libgd-dev
-
 install:
+  - mkdir ~/bin && ln -s $(which ccache) ~/bin/$CC && export PATH=$HOME/bin:$PATH
```
Comment (chipitsine) [inline .travis.yml]: using "ln" is recommended way, it is documented in "man ccache", anyway, we save more on openssl, than nginx

File: .travis.yml (hunk 1 above)
Comment (chipitsine) [inline .travis.yml]: I've got several "unused option -I...." like described here http://stackoverflow.com/questions/13238511/

Comment (agentzh) [inline .travis.yml]: @chipitsine Will you document this as a code comment around this line, maybe?

File: .travis.yml (hunk 2 above)
Comment (agentzh) [inline .travis.yml]: @chipitsine I reread the manpage of ccache and I failed to see any propounding reason of using the masquerading approach other than convenience when ccache'ing all the compilation in the current session. Well, given that Travis CI is using a VM approach and invasive settings are common practice, so maybe it also makes things more convenient if not really elegant.

Comment (chipitsine) [inline .travis.yml]: yes, if we will want to get rid of ccache, we can just remove that line

Comment (agentzh) [inline .travis.yml]: @chipitsine I'm fine with that.

**Comment by chipitsine** [comment]
it seems ccache is not stable in case of openssl (sometimes it helps to remove "-j" option in make)
I'll need time to investigate it

**Comment by agentzh** [comment]
@chipitsine I'm not feeling comfortable with removing `-jN` from openssl's `make` command. It looks like something wrong with ccache. And if ccache is not really trust worthy, there is a potential risk of using it in our Travis CI builds.

**Comment by chipitsine** [comment]
I will live just "apt-get" cache for now. Will remove ccache (and will investigate it later)

**Comment by agentzh** [comment]
@chipitsine Sounds good to me :)

**Comment by agentzh** [comment]
Merged to master. Thanks!

**Pull request closed without merging**